db/columns/operations/alter.py

```diff
@@ -103,12 +103,22 @@ def retype_column(
     except DataError as e:
         if type(e.orig) == InvalidParameterValue:
             raise InvalidTypeOptionError
-        if type(e.orig) == InvalidTextRepresentation:
-            raise InvalidTypeError
+        elif type(e.orig) == InvalidTextRepresentation:
+            raise InvalidTypeError(column_name, new_type)
+        elif type(e.orig) == StringDataRightTruncation:
```

Comment:
I think this is specific and understandable enough to just pass along rather than using a custom exception.